library/core/src/sync/atomic.rs

```diff
@@ -2864,62 +2963,62 @@ macro_rules! atomic_int {
-            /// assert_eq!(some_var.compare_and_swap(5, 10, Ordering::Relaxed), 5);
-            /// assert_eq!(some_var.load(Ordering::Relaxed), 10);
-            ///
-            /// assert_eq!(some_var.compare_and_swap(6, 12, Ordering::Relaxed), 10);
-            /// assert_eq!(some_var.load(Ordering::Relaxed), 10);
-            /// ```
+            // ///
+            // /// **Note**: This method is only available on platforms that support atomic operations on
+            // #[doc = concat!("[`", $s_int_type, "`].")]
+            // ///
+            // /// # Migrating to `compare_exchange` and `compare_exchange_weak`
+            // ///
+            // /// `compare_and_swap` is equivalent to `compare_exchange` with the following mapping for
+            // /// memory orderings:
+            // ///
+            // /// Original | Success | Failure
+            // /// -------- | ------- | -------
+            // /// Relaxed  | Relaxed | Relaxed
+            // /// Acquire  | Acquire | Acquire
+            // /// Release  | Release | Relaxed
+            // /// AcqRel   | AcqRel  | Acquire
+            // /// SeqCst   | SeqCst  | SeqCst
+            // ///
+            // /// `compare_and_swap` and `compare_exchange` also differ in their return type. You can use
+            // /// `compare_exchange(...).unwrap_or_else(|x| x)` to recover the behavior of `compare_and_swap`,
+            // /// but in most cases it is more idiomatic to check whether the return value is `Ok` or `Err`
+            // /// rather than to infer success vs failure based on the value that was read.
+            // ///
+            // /// During migration, consider whether it makes sense to use `compare_exchange_weak` instead.
+            // /// `compare_exchange_weak` is allowed to fail spuriously even when the comparison succeeds,
+            // /// which allows the compiler to generate better assembly code when the compare and swap
+            // /// is used in a loop.
+            // ///
+            // /// # Examples
+            // ///
+            // /// ```
+            // #[doc = concat!($extra_feature, "use std::sync::atomic::{", stringify!($atomic_type), ", Ordering};")]
+            // ///
+            // #[doc = concat!("let some_var = ", stringify!($atomic_type), "::new(5);")]
+            // ///
+            // /// assert_eq!(some_var.compare_and_swap(5, 10, Ordering::Relaxed), 5);
+            // /// assert_eq!(some_var.load(Ordering::Relaxed), 10);
+            // ///
+            // /// assert_eq!(some_var.compare_and_swap(6, 12, Ordering::Relaxed), 10);
+            // /// assert_eq!(some_var.load(Ordering::Relaxed), 10);
+            // /// ```
             #[inline]
             #[$stable]
             #[deprecated(
                 since = "1.50.0",
-                note = "Use `compare_exchange` or `compare_exchange_weak` instead")
-            ]
+                note = "Use `compare_exchange` or `compare_exchange_weak` instead"
+            )]
```

Comment:
Please no formatting
```suggestion
                note = "Use `compare_exchange` or `compare_exchange_weak` instead")
            ]
```

Reply:
Why is that module so badly formatted though? Is upstream not running tidy tests for it?